gadgets/trace_open/test/trace_open_test.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2019-2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"flag"
+	"io/fs"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/inspektor-gadget/inspektor-gadget/integration"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/command"
+	igrunner "github.com/inspektor-gadget/inspektor-gadget/pkg/testing/ig"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/match"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+type Event struct {
+	eventtypes.CommonData
+
+	// Type indicates the kind of this event
+	Type string `json:"type"`
+
+	// Message when Type is ERR, WARN, DEBUG or INFO
+	Message string `json:"message,omitempty"`
+}
+
+type traceOpenEvent struct {
+	Event
+
+	MountNsID uint64      `json:"mountnsid"`
+	Pid       uint32      `json:"pid"`
+	Uid       uint32      `json:"uid"`
+	Gid       uint32      `json:"gid"`
+	Comm      string      `json:"comm"`
+	Ret       int         `json:"ret"`
+	Err       int         `json:"err"`
+	Flags     int         `json:"flags"`
+	FlagsRaw  int32       `json:"flagsRaw"`
+	Mode      int         `json:"mode"`
+	ModeRaw   fs.FileMode `json:"modeRaw"`
+	FName     string      `json:"fname"`
+	FullPath  string      `json:"fullPath"`
+}
+
+var gadgetTest = flag.Bool("gadgets-test", false, "run gadget tests")
```

Comment:
I'm wondering how we can avoid doing this for all gadgets 🤔 ... Probably creating a pkg in `gadgets/testing` that defines this flag and implements a `RunGadgetTests` function that checks the value. Something like this:

`gadgets/testing/testing.go` (Not in `pkg/testing` because this is something we don't want to expose to users but it's an internal requirement for the way we run our tests from the makefiles/ci):
```go
package testing

import "flag"

var gadgetTest = flag.Bool("gadgets-test", false, "run gadget tests")

func RunGadgetTests() bool {
	return *gadgetTest
}
```

`gadgets/trace_open/test/trace_open_test.go`:
```go
import (
	...
	. "github.com/inspektor-gadget/inspektor-gadget/gadgets/testing"
	...
)

func TestTraceOpen(t *testing.T) {
	if !RunGadgetTests() {
		t.Skip()
	}
	...
}
```
 
Anyway, let's manage this in another PR where we add other tests.

@alban, any comment here?

Reply:
> Why do we need this option -gadgets-test in the first place?
>
I guess Pranav get inspiration from what we do for `integration-tests`:

Call:

https://github.com/inspektor-gadget/inspektor-gadget/blob/abd4dfda26fec0def8735bc2f1de21b9c51c84c2/Makefile#L280

Usage:

https://github.com/inspektor-gadget/inspektor-gadget/blob/abd4dfda26fec0def8735bc2f1de21b9c51c84c2/integration/inspektor-gadget/integration_test.go#L102-L105

But, I'm fine with the `os.Getenv("IG")` approach. It confirmed in [this discussion](https://github.com/inspektor-gadget/inspektor-gadget/pull/2607#discussion_r1548093241) that we should avoid exporting things in the Makefile and only pass env variables to the commands that use them.

Reply:
It would be better to make this change in this PR itself, will try to update the test in test commit itself.